src/odemis/acq/move.py

```
@@ -1861,6 +1863,186 @@ def _transformFromChamberToStage(self, shift: Dict[str, float]) -> Dict[str, flo
+        elif current_position == FM_IMAGING:
+            if target_pos_lbl == GRID_1:
+                end_pos = self._transformFromSEMToMeteor(get_sem_grid(GRID_1))
+            elif target_pos_lbl == GRID_2:
+                end_pos = self._transformFromSEMToMeteor(get_sem_grid(GRID_2))
+            elif target_pos_lbl == SEM_IMAGING:
+                end_pos = self._transformFromMeteorToSEM(self.stage.position.value)
+
+        if end_pos is None:
+            raise ValueError(
+                f"Unknown target position {POSITION_NAMES.get(target_pos_lbl, target_pos_lbl)} "
+                f"when in {POSITION_NAMES.get(current_position, current_position)}")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)